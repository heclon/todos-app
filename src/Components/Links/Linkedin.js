import React, { Component } from "react";
class Linkedin extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <a rel="noopener noreferrer" href={this.props.link} target="_blank">
        {this.props.author}
      </a>
    );
  }
}

export default Linkedin;
