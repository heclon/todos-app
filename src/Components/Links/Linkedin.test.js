import React from "react";
import { mount } from "enzyme";
import Linkedin from "./Linkedin";
import renderer from "react-test-renderer";

describe("Linkedin renders correctly", () => {
  beforeEach(() => {});

  it("renders the name linking to linkedin page", () => {
    const wrapper = mount(
      <Linkedin author="John Doe" link="https://linkedin.com/in/john.doe" />
    );
    expect(wrapper.text()).toEqual("John Doe");
  });

  it("snapshot test render correctly App component", () => {
    const wrapper = renderer
      .create(
        <Linkedin author="John Doe" link="https://linkedin.com/in/john.doe" />
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
