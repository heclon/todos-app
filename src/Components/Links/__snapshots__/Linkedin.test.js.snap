// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Linkedin renders correctly snapshot test render correctly App component 1`] = `
<a
  href="https://linkedin.com/in/john.doe"
  rel="noopener noreferrer"
  target="_blank"
>
  John Doe
</a>
`;
